### PR TITLE
chore: update changelog

### PR DESCRIPTION
**Pull request by crodriguezvega**
## Description

- Updates unreleased section with the content from the automatic changelog generator (thanks @chatton for generating it!).
- Adds changelog for v4.3.0, v5.2.0 and v6.1.0 that was missing.

closes: #XXXX

### Commit Message / Changelog Entry

```bash
type: commit message
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules/10-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [x] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

**Review by chatton — COMMENTED** [review]
Nice it looks good to me! 

I think we are missing an entry for the overall 02-client refactor (w/link to docs)

There are a few entries I don't think we need to include in the changelog, let me know if you disagree!

File: CHANGELOG.md [hunk 1]
```diff
@@ -61,12 +63,15 @@ Ref: https://keepachangelog.com/en/1.0.0/
 * (06-solomachine) [\#2761](https://github.com/cosmos/ibc-go/pull/2761) Removed deprecated `ClientId` field from `Misbehaviour` and `allow_update_after_proposal` field from `ClientState`.
 * (apps) [\#3154](https://github.com/cosmos/ibc-go/pull/3154)  Remove unused `ProposalContents` function.
 * (apps) [#3149](https://github.com/cosmos/ibc-go/pull/3149) Remove legacy interface function `RandomizedParams`, which is no longer used.
-
+* (light-clients/08-solomachine) [#2941](https://github.com/cosmos/ibc-go/pull/2941) Removing solomachine header sequence.
```
Comment (colin-axner) [inline CHANGELOG.md:66]:
```suggestion
* (light-clients/06-solomachine) [#2941](https://github.com/cosmos/ibc-go/pull/2941) Removing solomachine header sequence.
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -112,12 +126,36 @@ Ref: https://keepachangelog.com/en/1.0.0/
...
 * (modules/core/02-client)[\#1676](https://github.com/cosmos/ibc-go/pull/1676) ClientState must be zeroed out for `UpgradeProposals` to pass validation. This prevents a proposal containing information governance is not actually voting on.
 * (core/02-client) [\#2510](https://github.com/cosmos/ibc-go/pull/2510) Fix client ID validation regex to conform closer to spec.
+* (apps/transfer) [#3045](https://github.com/cosmos/ibc-go/pull/3045) Allow value with slashes in URL template.
+* (apps/27-interchain-accounts) [#2601](https://github.com/cosmos/ibc-go/pull/2601) Remove bech32 check from owner address on ICA controller msgs RegisterInterchainAccount and SendTx.
+* (apps/transfer) [#2651](https://github.com/cosmos/ibc-go/pull/2651) Skip emission of unpopulated memo field in ics20.
+* (apps/27-interchain-accounts) [#2682](https://github.com/cosmos/ibc-go/pull/2682) Avoid race conditions in ics27 handshakes.
+* (light-clients/08-solomachine) [#2741](https://github.com/cosmos/ibc-go/pull/2741) Added check for empty path in 06-solomachine.
```
Comment (colin-axner) [inline CHANGELOG.md:137]: all instance of `08-solomachine` should be `06-solomachine`

File: CHANGELOG.md [hunk 3]
```diff
@@ -112,12 +126,36 @@ Ref: https://keepachangelog.com/en/1.0.0/
...
+* (light-clients/07-tendermint) [#3022](https://github.com/cosmos/ibc-go/pull/3022) Correctly close iterator in `07-tendermint` store.
+* (core/02-client) [#3010](https://github.com/cosmos/ibc-go/pull/3010) Update `Paginate` to use `FilterPaginate` in `ClientStates` and `ConnectionChannels` grpc queries.
+
+### Refactor
+ 
+* (light-clients/08-solomachine) [#1972](https://github.com/cosmos/ibc-go/pull/1972) Panic on ZeroCustomFields for solo machine implementation.
+* (core/02-client) [#2864](https://github.com/cosmos/ibc-go/issues/2864) Simplify automatic migration code by using client keeper functions.
+* (light-clients/07-tendermint) [#2862](https://github.com/cosmos/ibc-go/issues/2862) Simplify optional tendermint pruning migrations.
```
Comment (colin-axner) [inline CHANGELOG.md:145]: I think these could be removed. It was simplifying a feature that was added/unreleased

**Review by colin-axner — APPROVED** [review]

File: CHANGELOG.md (hunk 2 above)
Comment (crodriguezvega) [reply CHANGELOG.md]: I don't know what I was thinking... 😆

**Pull request merged**